python/ray/train/torch/torch_predictor.py

```diff
@@ -21,17 +24,42 @@ class TorchPredictor(Predictor):
         model: The torch module to use for predictions.
         preprocessor: A preprocessor used to transform data batches prior
             to prediction.
+        use_gpu: If set, the model will be moved to GPU on instantiation and
+            prediction happens on GPU.
     """
 
     def __init__(
-        self, model: torch.nn.Module, preprocessor: Optional["Preprocessor"] = None
+        self,
+        model: torch.nn.Module,
+        preprocessor: Optional["Preprocessor"] = None,
+        use_gpu: bool = False,
     ):
         self.model = model
+        self.model.eval()
         self.preprocessor = preprocessor
 
+        # TODO (jiaodong): #26249 Use multiple GPU devices with sharded input
+        self.use_gpu = use_gpu
+        if use_gpu:
+            # Ensure input tensor and model live on GPU for GPU inference
+            self.model.to(torch.device("cuda"))
+
+        if use_gpu is False and torch.cuda.device_count() > 0:
```

Comment:
Can we only log this once (use `ray.util.log_once` utility)? So if we do batch prediction we don't see this printed out a bunch of times.